src/foam/nanos/fs/File.js

```diff
@@ -298,5 +300,40 @@ foam.CLASS({
         return "";
       `
     }
+  ],
+
+  actions: [
+    {
+      name: 'download',
+      code: function(a, X) {
+        // TODO: Add logging for who has downloaded files etc.
+        var blob = this.data;
+        if ( foam.blob.BlobBlob.isInstance(blob) ) {
+          window.open(URL.createObjectURL(blob.blob));
+        } else {
+          var url = this.address;
+          window.open(url);
+        }
+      }
+    },
+    {
+      name: 'multiDownload',
+      label: 'Download',
+      isAvailable: function() { return false; },
+      code: function(ctx, X) {
+        // For multi-select download support in DAOBrowserView
+        if ( this.config && this.config.selectedObjs && ! foam.Object.equals(this.config.selectedObjs, {}) ) {
+          foam.Object.forEach(this.config.selectedObjs, function(obj) {
```

Comment:
Since this is meant to only be rendered in DAOBrowserView `this` here would be the DAOBrowserView. I think I might have removed some code from there during my cleanup. I've added it back now so it should make sense